app/Http/Controllers/UploadRaceController.php

```diff
@@ -30,98 +26,76 @@ public function index()
         ]);
     }
 
-    /**
-     * Show the form for creating a new resource.
-     */
-    public function create()
-    {
-        //
-    }
-
-    /**
-     * Store a newly created resource in storage.
-     */
     public function store(Request $request)
     {
-        // here we calculate the time in thousants so we only have to store whole numbers in the database and need only one column
+        // here we calculate the time in thousands so we only have to store whole numbers in the database and need only one column
         $time = intval($request['minutes']) * 60 + intval($request['seconds']) + intval($request['thousands']) / 1000;
         $userId = Auth::user()->id;
         $raceId = $request['race_id'];
         $extension = $request->controlPicture->extension();
         $allExtensions = ['jpg', 'png', 'jpeg', 'svg'];
-        // check if the mime type is valid
-        if (in_array(strtolower($extension), $allExtensions)) {
-            $raceResult = RaceResult::where('user_id', $userId)->where('race_id', $raceId)->first();
-            if ($raceResult) {
-                if ($raceResult->seconds < $time) {
-                    return back()->withErrors(['de tijd die je probeert te uploaden is langzamer dan je huidige tijd'])->withInput();
-                }
-                $raceResult->update([
-                    'user_id' => $userId,
-                    'seconds' => $time,
-                    'is_valid' => false,
-                ]);
 
-            } else {
-                $raceResult = RaceResult::create([
-                    'user_id' => $userId,
-                    'race_id' => $raceId,
-                    'seconds' => $time,
-                    'is_valid' => false,
-                ]);
-            }
-            DB::table('race_results')
-                ->where('id', $raceResult->id)
-                ->update(['picture_name' => $raceResult->id . '.' . $extension]);
+        // validate if file mime type is valid
+        if (!in_array(strtolower($extension), $allExtensions)) {
+            return $this->redirectBackWithErrorMimeTypeInvalid();
+        }
 
-            return view('home', [
-                'user' => Auth::user()
+        // retrieve the race raceresult
+        $raceResult = RaceResult::where('user_id', $userId)
+            ->where('race_id', $raceId)
+            ->first();
+
+        // When the race result is already existant we want to update it to prevent double records from the same user.
```

Comment:
this is a problem, I made the controller so that all the results would be stored in the db so that the user can see him/her self improving.

Reply:
word nu aangepast